npm/simple-mock.js

```diff
@@ -0,0 +1,914 @@
+                '<!DOCTYPE html><html><head><title>Hello World!</title></head><body><h1>Hello World!</h1></body></html>';
+            }
+            else if (uri.substring(6, 10) === 'json') {
+                resHeader['Content-Type'] = 'application/json; charset=utf-8';
+                body = {
+                    alpha: {
+                        beta: {
+                            gamma: true,
+                            delta: [],
+                            epsilon: {},
+                            phi: 1.618,
+                            pi: 3.14
+                        }
+                    }
+                };
+            }
+
+            return [200, body, resHeader];
+        });
+}
+
+function mockGZIPendpoint () {
+    var allowUnmocked = { allowUnmocked: true };
+
+    nock('https://postman-echo.com', allowUnmocked)
+        .persist()
+        // .log(console.log)
+        .get('/gzip')
+        .reply(function () {
+            return [
+                200,
+                {
+                    gzipped: true,
+                    headers: {
+                        'x-forwarded-proto': 'https',
+                        host: 'postman-echo.com',
+                        accept: '*/*',
+                        'accept-encoding': 'gzip, deflate',
+                        'cache-control': 'no-cache',
+                        cookie:
+                        // eslint-disable-next-line max-len
+                        'sails.sid=s%3AzVZKZEAc1IvL61qZyqrR2MSRO49mI9bc.iVKWR9D0pzmWZf158wNUp6thKfAHKQRU4JTUxnKFurc; foo=bar',
+                        'postman-token': '1c51c9ac-e4b4-40c0-a821-9317d70d9dc1',
+                        'user-agent': 'PostmanRuntime/7.6.0',
+                        'x-forwarded-port': '443'
+                    },
+                    method: 'GET'
+                },
+                this.req.headers
+            ];
+        });
+}
+
+function mockRedirects () {
+    var allowUnmocked = { allowUnmocked: true };
+
+    nock(/https:\/\/.*.com\//, allowUnmocked)
+        .persist()
+        // .log(console.log)
+        .get(/\/redirect-to.*/)
+        // .query(true)
+        .reply(function () {
+            return [302, {
+                args: {},
+                headers: {
+                    'x-forwarded-proto': 'https',
+                    host: 'postman-echo.com',
+                    accept: '*/*',
+                    'accept-encoding': 'gzip, deflate',
+                    'cache-control': 'no-cache',
+                    cookie:
+                    // eslint-disable-next-line max-len
+                    'sails.sid=s%3Amz8FY8zkbUSVOHUm1vV6dhz9yi7vpanU.zbJ%2FdhTOwNu%2BKelkJqM8LF1rODiONamyejECgbnEmig; foo=bar',
+                    'postman-token': 'c4467e84-85cd-4ea1-bfa4-5d0ddf855025',
+                    referer: 'https://postman-echo.com/redirect-to?url=https://postman-echo.com/get',
+                    'user-agent': 'PostmanRuntime/7.6.0',
```

Comment:
This would cause tests to break when the version of runtime gets bumped, see how this has been handled in other tests.